test/helpers/patient_query_helper_test.rb

```diff
@@ -258,6 +258,75 @@ class PatientQueryHelperTest < ActionView::TestCase
+
+    patients = Patient.all
+    tz_offset = 300
+
+    # Check for monitorees who have a blank foreign address
+    filters = [{ filterOption: {}, value: [{ name: 'address-foreign', value: '' }] }]
+    filters[0][:filterOption]['name'] = 'address'
+    filtered_patients = advanced_filter(patients, filters, tz_offset)
+    filtered_patients_array = [patient_1, patient_2, patient_3, patient_8, patient_9, patient_10]
+    assert_equal filtered_patients_array.pluck(:id), filtered_patients.pluck(:id)
+
+    # Check for monitorees who have a foreign address
+    filters = [{ filterOption: {}, value: [{ name: 'address-foreign', value: '123 main st' }] }]
+    filters[0][:filterOption]['name'] = 'address'
+    filtered_patients = advanced_filter(patients, filters, tz_offset)
+    filtered_patients_array = [patient_4, patient_6, patient_7, patient_11]
+    assert_equal filtered_patients_array.pluck(:id), filtered_patients.pluck(:id)
+
+    # Check for monitorees who have a blank domestic address
+    filters = [{ filterOption: {}, value: [{ name: 'address-usa', value: '' }] }]
+    filters[0][:filterOption]['name'] = 'address'
+    filtered_patients = advanced_filter(patients, filters, tz_offset)
+    filtered_patients_array = [patient_4, patient_5, patient_6, patient_8, patient_10, patient_11]
+    assert_equal filtered_patients_array.pluck(:id), filtered_patients.pluck(:id)
+
+    # Check for monitorees who have a foreign address
+    filters = [{ filterOption: {}, value: [{ name: 'address-usa', value: '123 main st' }] }]
+    filters[0][:filterOption]['name'] = 'address'
+    filtered_patients = advanced_filter(patients, filters, tz_offset)
+    filtered_patients_array = [patient_1, patient_3, patient_7, patient_9]
+    assert_equal filtered_patients_array.pluck(:id), filtered_patients.pluck(:id)
+
+    # Check for monitorees who have both a domestic and foreign address
+    filters = [
+      { filterOption: {}, value: [{ name: 'address-usa', value: '123 main st' }] },
+      { filterOption: {}, value: [{ name: 'address-foreign', value: '123 main st' }] }
+    ]
+    filters[0][:filterOption]['name'] = 'address'
+    filters[1][:filterOption]['name'] = 'address'
+    filtered_patients = advanced_filter(patients, filters, tz_offset)
+    filtered_patients_array = [patient_7]
+    assert_equal filtered_patients_array.pluck(:id), filtered_patients.pluck(:id)
+
+    # Check for monitorees who have neither a domestic nor foreign address
+    filters = [
+      { filterOption: {}, value: [{ name: 'address-usa', value: '' }] },
+      { filterOption: {}, value: [{ name: 'address-foreign', value: '' }] }
+    ]
+    filters[0][:filterOption]['name'] = 'address'
+    filters[1][:filterOption]['name'] = 'address'
+    filtered_patients = advanced_filter(patients, filters, tz_offset)
+    filtered_patients_array = [patient_8, patient_10]
+    assert_equal filtered_patients_array.pluck(:id), filtered_patients.pluck(:id)
+  end
+
```

Comment:
thoughts on adding tests for number type as well?

Reply:
https://github.com/SaraAlert/SaraAlert/pull/1252/commits/139bc7d564fd6b8256a9c0f121518989e083bac6